File: test_websocket.py
import asyncio
import websockets

async def test_websocket():
    uri = "ws://localhost:5000/ws"

    async with websockets.connect(uri) as websocket:
        try:
            # Send a message
            message = "what is the temperature in New York city"
            await websocket.send(message)
            print(f"Sent: {message}")

            # Receive response
            response = await websocket.recv()
            print(f"Received: {response}")

        except websockets.ConnectionClosed as e:
            print(f"Connection closed: {e}")

        except Exception as e:
            print(f"An error occurred: {e}")

if __name__ == "__main__":
    asyncio.run(test_websocket())

File: testWebSocket.js
const io = require('socket.io-client');

// Connect to the WebSocket server (Socket.IO)
const socket = io('http://localhost:5000', {
  transports: ['websocket'],
});

socket.on('connect', () => {
  console.log('Connected to the WebSocket server');
  // Send a test message
  socket.emit('message', "Hello, I want to know the weather in New York");
});

socket.on('message', (data) => {
  console.log('Received message:', data);
  // Add additional logic to handle different types of messages
});

socket.on('disconnect', () => {
  console.log('Disconnected from WebSocket server');
});

socket.on('error', (error) => {
  console.error('Error:', error);
});
